main.tf

```diff
@@ -18,10 +18,24 @@ resource "azurerm_mssql_server" "vaconfigured" {
   version                      = "12.0"
   administrator_login          = "4dm1n157r470r"
   administrator_login_password = "4-v3ry-53cr37-p455w0rd"
-  minimum_tls_version          = "1.1"
+  minimum_tls_version          = "1.2"
 
   transparent_data_encryption_key_vault_key_id = azurerm_key_vault_key.vaconfigured.id
 
+
+  extended_auditing_policy {
```

Comment:
**Tenable.cs remediation**
 
**[MEDIUM]: Ensure that public access is disabled in Azure Key Vault**
 
**Vulnerability**
Allowing unrestricted, public access to cloud services could open an application up to external attack. Disallowing this access is typically considered best practice. For a service such as Azure Key Vault, disallowing this access also helps protect the services that utilize keys, secrets, and certificates. In addition, this can help a company adhere to compliance and regulatory requirements.
 
**Remediation**
In Azure Console -
1. Open the Azure Portal and go to Key Vaults.
2. Choose the Key vault you wish to edit.
3. Under Settings, select Networking.
4. On the Firewalls and virtual networks tab, set 'Allow access from' to the 'Disable public access' option.

In Terraform - 
1. In the azurerm_key_vault resource, set the public_network_access_enabled field to false.

References:
[https://learn.microsoft.com/en-us/azure/key-vault/general/overview](https://learn.microsoft.com/en-us/azure/key-vault/general/overview)
[https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/key_vault](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/key_vault)

source: tenablecs-